nautobot/ui/src/views/generic/ObjectList.jsx

```diff
@@ -1,24 +1,77 @@
-import { useParams } from "react-router-dom"
+import { useParams, useLocation, useSearchParams } from "react-router-dom"
+import { Text } from "@nautobot/nautobot-ui"
+import { LoadingWidget } from "@components/common/LoadingWidget"
 
-import ListViewTemplate from "@views/BSListViewTemplate"
+import ObjectListTable from "@components/common/ObjectListTable"
 
 
-export default function BSListView({ list_url }) {
+import useSWR from "swr"
+const fetcher = (...urls) => {
+  const f = url => fetch(url, { credentials: "include" }).then(r => r.json()).catch(r => r)
+  return Promise.all(urls.map(url => f(url)))
+}
+
+export default function GenericObjectListView() {
   const { app_name, model_name } = useParams()
-  // TODO: Restore page titles
-  // let pagetitle = "Nautobot"
-  // if (model_name) {
-  //   pagetitle = model_name.replace("-", " ")
-  //   pagetitle = pagetitle[0].toUpperCase() + pagetitle.slice(1)
-  // }
+  const { [0]: searchParams } = useSearchParams();
+
+  // TODO: If we can solve the dynamic API of RTK-Query we won't need to construct the API URLs
+  const list_url = `/api/${app_name}/${model_name}/`
+  const headers_url = list_url + "table-fields/"
```

Comment:
There's also the `useLocation()` effect hook, and I'm wondering if we might consider using that to grab the URI vs. manually constructing it? Not a big deal, but trying to think of ways to avoid having to hard-code URLs.

Reply:
We shouldn't be crafting this in this way. This is overall just ugly. We should provide a getter that doesn't have to construct the URLs based on an assumption on how the API is constructed underneath.

Reply:
Even better. 😗